frontend/public/components/secrets/create-secret.tsx

```diff
@@ -927,14 +927,46 @@ class SecretLoadingWrapper extends React.Component<SecretLoadingWrapperProps, Se
   }
 }
 
-export const CreateSecret = ({match: {params}}) => {
-  const SecretFormComponent = secretFormFactory(params.type);
-  return <SecretFormComponent fixed={{ metadata: { namespace: params.ns } }}
-    secretTypeAbstraction={params.type}
-    explanation={secretFormExplanation[params.type]}
-    titleVerb="Create"
-    isCreate={true}
-  />;
+export class CreateSecret extends React.Component<CreateSecretProps, CreateSecretState> {
+  readonly state: CreateSecretState = {
+    formComponent: null,
+    secretTypeAbstraction: SecretTypeAbstraction.generic,
+  };
+  componentDidMount() {
+    const secretTypeAbstraction = this.props.match.params.type;
+    this.setState({
+      formComponent: secretFormFactory(secretTypeAbstraction),
+      secretTypeAbstraction,
+    });
+  }
+  render() {
+    const { secretTypeAbstraction, formComponent } = this.state;
+    const { params } = this.props.match;
+    if (!formComponent) {
```

Comment:
I think we can remove this check? If it's set in the constructor, it should always be set here (as long as we default to key/value for unknown types).